main.cpp

```diff
@@ -53,104 +53,34 @@ Project 4: Part 4 / 9
  You will need to use Forward Declaration and out-of-class definitions to complete this.
  */
 
+#include <iostream>
+#include <cmath>
+
+// Forward declarations for UDT calls
+struct FloatType;
+struct IntType;
+struct DoubleType;
+
 
 
 struct Point
 {
-    Point& multiply(float m)
-    {
-        x *= m;
-        y *= m;
-        return *this;
-    }
+    Point(float x_, float y_);
+    Point(const FloatType& x_, const FloatType& y_);
+    Point(const DoubleType& x_, const DoubleType& y_);
+    Point(const IntType& x_, const IntType& y_);
+
+    void toString();
+
+    Point& multiply(float m);
+    Point& multiply(FloatType& m);
+    Point& multiply(DoubleType& m);
+    Point& multiply(IntType& m);
 private:
     float x{0}, y{0};
+    Point& multiplyInternal(float m);
```

Comment:
You don't need a `multiplyInternal`.  just call `multiply(float)` from within the other functions
- [ ] revise